src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -46,92 +47,144 @@ const DiffExprCompute = (props) => {
   useEffect(() => {
     if (hierarchy.length === 0) return;
 
-    const newSelectableClusters = _.cloneDeep(hierarchy);
-    // create a new item for each hierarchy to represent All
-    // eslint-disable-next-line array-callback-return
-    newSelectableClusters.map(({ key, children }) => {
-      if (children && children.length > 0) {
-        children.unshift(generateSpecialKey(key));
+    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
+      if (!cellSetKey) {
+        return null;
       }
-    });
-    setSelectableClusters(newSelectableClusters);
-    setSelectedCellSets(_.mapValues(selectedCellSets, (cellSetKey) => {
-      if (isKeySpecial(cellSetKey)) {
-        return 'All';
+
+      const splitKey = cellSetKey.split('/')[1];
+
+      if (cellSetKey === 'all' || cellSetKey === 'background' || splitKey === 'rest') {
+        return cellSetKey;
       }
-      if (cellSetKey !== defaultSelected && !properties[cellSetKey]) {
-        return defaultSelected;
+
+      if (!properties[splitKey]) {
+        return null;
       }
+
       return cellSetKey;
     }));
   }, [hierarchy, properties]);
 
   const validateForm = () => {
-    if (selectedCellSets.cellSet === defaultSelected) {
+    if (!selectedGroups.cellSet) {
       setIsFormValid(false);
       return;
     }
-    if (selectedCellSets.compareWith === defaultSelected) {
+
+    if (!selectedGroups.compareWith) {
       setIsFormValid(false);
       return;
     }
-    if (selectedCellSets.cellSet === selectedCellSets.compareWith) {
+
+    if (!selectedGroups.basis) {
       setIsFormValid(false);
       return;
     }
+
     setIsFormValid(true);
   };
 
+  // Validate form when the groups selected changes.
   useEffect(() => {
     validateForm();
-  }, [selectedCellSets]);
+  }, [selectedGroups]);
+
+  // Erase selection when a different type of comparison is selected.
+  useEffect(() => {
+
```

Comment:
does this empty `useEffect` actually erase state like this?

Reply:
no, that was leftover code, fixed.